### PR TITLE
features: host override, dynamic pull and commit msg body

### PR DESCRIPTION
**Pull request by shcarrico**
This is a somewhat opinionated pull. 😺 

TLDR, this makes it possible to run just `santari --token your_token`. (You can also still use environment variable as before). All other required info is derived from your projects `package.json` by default.

It also creates dynamic PR and commit messages by using a diff of the existing vs updated dependency lists.

* read repo / host from package.json if not specified
  * will read package.json and parse out the github host, repository by default
  * can also add `repo` to your config file, or specify on cli as before
* accept api token on cli as `--token` as well as `GITHUB_KEY` environment variable
* will attempt to read a local `.santari.json` without specifying via `--config, -c` first, then will look for argument path
* merges config file props with cli so props can be specified similarly
  * defaults are picked first (mostly null except for the `GITHUB_KEY` env variable, then config file props, then cli args to allow for override, ie, the CLI arguments always win, then config file, then environment var.
* accept command line or config file arguments for similar props
  * updated cli yargs to follow convention for aliases and work similarly for all props by adding aliases and using `.option` call
* generates pull request body and commit message body dynamically based on diff of what was updated
* documentation updates to match above
* relaxes a few onerous eslint rules

example commit message:
```
 chore(package.json): update dependencies

Dev dependencies updated:
* `@types/jasmine : ^3.3.4 -> ^3.3.5`
* `@types/node : ^10.12.12 -> ^10.12.18`
```

example pull request body:

```
Dev dependencies updated:

    @types/jasmine : ^3.3.4 -> ^3.3.5
    @types/node : ^10.12.12 -> ^10.12.18
```

### CODE REVIEW

**Comment by jeremyrajan** [comment]
Thanks for the PR @shcarrico 😄 . I was actually planning on adding messages in the PR body, since you did it, its wonderful 🎉 . I will give it a shot later today and merge it!

File: src/libs/utils.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
 module.exports = {
   readConfigFile(file) {
+    // try to load package.json from execution context directory
+    const pkg = fs.readJSONSync(path.join(process.cwd(), 'package.json'), { throws: false });
+    let host, repo;
+
+    if (pkg) {
+      const pkgRepo = pkg.repository.url || pkg.repository;
+      const pkgGit = gitUrl(pkgRepo);
+      host = `${pkgGit.host}/api/v3`;
```
Comment (jeremyrajan) [inline src/libs/utils.js]: I dont think, we need to override it. Because normally the API path should be `api.github.com`, unless you setup the host in `.santari.json`.

File: .santari.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
   "pr": {
-    "title": "Updating Dependencies from file",
-    "body": "Dependencies to Update"
-  }
-}
\ No newline at end of file
+    "title": "PR title",
+    "body": "PR Body"
```
Comment (jeremyrajan) [inline .santari.json]: We can remove `title` and `body` from demo config :). Atleast we can show the pr body being populated.

**Comment by jeremyrajan** [comment]
hey @shcarrico, apologies in delay in going thru this. Everything looks good, except few changes I put in comments :). I tried to update it, but dont have access to the repo you are PR'ing from. Would be nice you can update and I can merge 🎉  and also resolve conflicts 👼

**Comment by jeremyrajan** [comment]
Example here: https://github.com/jeremyrajan/json-markdown/pull/23

File: src/tasks/santari.js [hunk 3]
```diff
@@ -17,30 +19,38 @@ const utils = require('../libs/utils');
...
+    this.repoDetails = this.client.repo(config.repo);
     this.masterSHA = ''; // master SHA
     this.packageSHA = ''; // package JSON SHA
     this.packagePath = ''; // package path from repo
     this.packageJSON = ''; // JSON parsed version of repo package
     this.packageTempPath = ''; // package path where temp package is stored. To be used by ncu
     this.depBranchName = `update-deps-santari-${Math.ceil(Math.random() * 100000)}`;
-    this.mainBranch = 'master';
+    this.mainBranch = config.branch || 'master';
+    this.commitMsg = config.msg || 'chore(package.json): update dependencies\n';
```
Comment (jeremyrajan) [inline src/tasks/santari.js]: May be we can use a different key, like `commitMessage` ?

**Comment by jeremyrajan** [comment]
@shcarrico did you get some time to update this? :)
